src/sentry/static/sentry/app/components/organizations/multipleEnvironmentSelector.jsx

```diff
@@ -28,94 +35,174 @@ class MultipleEnvironmentSelector extends React.Component {
     value: PropTypes.array,
   };
 
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
     this.state = {
       isOpen: false,
+      selectedEnvs: new Set(props.value),
     };
   }
 
-  handleUpdate = actions => {
-    let {value, onUpdate} = this.props;
+  /**
+   * If value in state is different than value from props, propagate changes
+   */
+  doChange = (value, e) => {
+    const {onChange} = this.props;
+
+    if (typeof onChange === 'function') {
```

Comment:
Do you need the additional typecheck if you have the PropType set?

Reply:
Nope